BuiltDifferentMobileApp/BuiltDifferentMobileApp/ViewModels/Coach/CoachDashboardPageViewModel.cs

```diff
@@ -66,19 +66,88 @@ public class CoachDashboardPageViewModel : ViewModelBase {
             List<Models.Client> clients = await networkService.GetAsync<List<Models.Client>>(APIConstants.GetClientsForCoachId(coachId));
 
             if(clients == null) {
-                Clients = new ObservableRangeCollection<Models.Client>();
+                originalClients = null;
+                ClientsOnBoard = new ObservableRangeCollection<Models.ClientWithProgress>();
             }
             else if(clients.Count != 0) {
-                originalClients = clients;
-                Clients = new ObservableRangeCollection<Models.Client>(clients);
+
+                List<Models.ClientWithProgress> orig = new List<Models.ClientWithProgress>();
+                List<Models.ClientWithProgress> onBoard = new List<Models.ClientWithProgress>();
+
+                foreach (Models.Client client in clients)
+                {
+                    int completedMeals = 0;
+                    int totalMeals = 0;
+                    int completedWorkouts = 0;
+                    int totalWorkouts = 0;
+
+                    List<Models.Workout> workoutsByClient = await networkService.GetAsync<List<Models.Workout>>(APIConstants.GetWorkoutsByClientId(client.id));
+                    if (workoutsByClient == null)
+                    {
+                        totalWorkouts = 0;
+                        completedWorkouts = 0;
+                    }
+                    else if(workoutsByClient.Count != 0)
+                    {
+                        totalWorkouts = workoutsByClient.Count(workout => (workout.day).ToString("yyyy-MM-dd") == DateTime.Now.Date.ToString("yyyy-MM-dd"));
+                        completedWorkouts = workoutsByClient.Count(workout => (workout.isCompleted == true) && ((workout.day).ToString("yyyy-MM-dd") == DateTime.Now.Date.ToString("yyyy-MM-dd")));
+                    }
+
+                    List<Models.Meal> mealsByClient = await networkService.GetAsync<List<Models.Meal>>(APIConstants.GetMealsByClientId(client.id));
+                    if (mealsByClient == null)
+                    {
+                        totalMeals = 0;
+                        completedMeals = 0;
+                    }else if (mealsByClient.Count != 0)
+                    {
+                        totalMeals = mealsByClient.Count(meal => (meal.day).ToString("yyyy-MM-dd") == DateTime.Now.Date.ToString("yyyy-MM-dd"));
+                        completedMeals = mealsByClient.Count(meal => (meal.isEaten == true) && ((meal.day).ToString("yyyy-MM-dd") == DateTime.Now.Date.ToString("yyyy-MM-dd")));
+                    }
+
+                    
+                    
+                    int totalTasks = totalWorkouts + totalMeals;
+                    int completedTasks = completedMeals + completedWorkouts;
+
+                    string percentage = "";
+                    if(totalTasks != 0)
+                    {
+                        percentage = ((completedTasks * 100) / totalTasks).ToString() + " %";
```

Comment:
Could be nice to cut this off after two decimals

Reply:
it is handled in the xaml with the String Format property of the binding. 
![image](https://user-images.githubusercontent.com/54686754/153049502-af912df5-b4d6-46a4-828c-d2aa1487b024.png)